src/coreclr/vm/methodtable.cpp

```diff
@@ -3795,6 +3795,24 @@ void MethodTable::EnsureStaticDataAllocated()
     pAuxiliaryData->SetIsStaticDataAllocated();
 }
 
+BOOL MethodTable::IsClassInitedOrPreinited()
+{
+    CONTRACTL
+    {
+        THROWS;
+        GC_TRIGGERS;
+        INJECT_FAULT(COMPlusThrowOM());
+    }
+    CONTRACTL_END;
+
+    if (IsClassInited())
+        return TRUE;
+    
+    AttemptToPreinit();
```

Comment:
`AttemptToPreinit` is a bit complex method. We may end up calling it here repeatedly number of times.

Would it better to set the initialized flag during `SetIsStaticDataAllocated` so that we do not need to be repeating the preinit check and all calls to `IsClassInited()` will take care of the pre-initialization automatically?

```
    inline void SetIsStaticDataAllocated()
    {
        LIMITED_METHOD_CONTRACT;
        DWORD dwFlags = enum_flag_IsStaticDataAllocated;
        if (IsPreinited())
            dwFlags |= enum_flag_Initialized;
        InterlockedOr((LONG*)&m_dwFlags, (LONG)dwFlags);
    }
```

Reply:
Mmm... seems like a decent idea, I'll look into that once I understand the arm64 failure seen in the tests.

Reply:
The linux arm64 failure is happening on multiple PRs. Opened https://github.com/dotnet/runtime/issues/104263.